### PR TITLE
[SYCL] Update C++ standard for DPC++ runtime to C++14

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
Signed-off-by: Dmitry Vodopyanov <dmitry.vodopyanov@intel.com>

### CODE REVIEW

File: sycl/doc/GetStartedGuide.md [hunk 1]
```diff
@@ -514,7 +514,7 @@ class CUDASelector : public cl::sycl::device_selector {
 
 ## C++ standard
 
-- Minimal supported C++ standard is C++11 on Linux and C++14 on Windows.
+- Minimal supported C++ standard is C++14 on Linux and Windows.
```
Comment (Ruyk) [inline sycl/doc/GetStartedGuide.md:517]: Is this referring to the input programs supported by the compiler? The SYCL 1.2.1 standard is based on C++11 so any C++11 program should be valid. Is there any reason why the minimal supported standard is C++14?
I think that is different from what toolchain or library version you need .

Comment (dm-vodopyanov) [reply sycl/doc/GetStartedGuide.md:517]: DPC++ applications are built as C++14 by default. We move DPC++ RT to C++14 standard as well to support required C++14 features in the RT.

Comment (Ruyk) [reply sycl/doc/GetStartedGuide.md:517]: Yeah I understand that, let me rephrase the question: Is there any reason why a valid C++11 program will not work on DPC++ after making this change? (is there even a valid C++11 program that is not valid C++14)
Does the DPC++ API or any of the extensions require a C++14 feature?
Is this the behavior and the minimum requirement even when the `sycl-std` flag is set to 1.2.1?

Comment (dm-vodopyanov) [reply sycl/doc/GetStartedGuide.md:517]: It's not recommended to build your C++11 code with C++14 library because of some cases, but it's possible. There is a good explanation about this: https://stackoverflow.com/a/38525723

> Does the DPC++ API or any of the extensions require a C++14 feature?

It's not for now. C++14 features like ```std::shared_timed_mutex``` are required in implementation of DPC++ RT internal classes.

> Is this the behavior and the minimum requirement even when the sycl-std flag is set to 1.2.1?

I assume yes, because C++14 is a standard by default in DPC++ Compiler, adding @mdtoguchi to comment.

Comment (mdtoguchi) [reply sycl/doc/GetStartedGuide.md:517]: The setting of `-sycl-std=1.2.1` (or any other setting except the use of `-std=`) does not change the DPC++ `-std=c++14` value

Comment (pvchupin) [reply sycl/doc/GetStartedGuide.md:517]: I believe we are not prohibiting use of -std=c++11 and this is too strong statement in our docs. What we really need to say is that:
* DPC++ RT is built as C++14 library 
* DPC++ compiler is building apps as C++14 apps by default.

@dm-vodopyanov, can you update text accordingly please?

Comment (dm-vodopyanov) [reply sycl/doc/GetStartedGuide.md]: Thanks, updated.

File: sycl/doc/GetStartedGuide.md [hunk 2]
```diff
@@ -514,7 +514,8 @@ class CUDASelector : public cl::sycl::device_selector {
 
 ## C++ standard
 
-- Minimal supported C++ standard is C++11 on Linux and C++14 on Windows.
+- DPC++ runtime is built as C++14 library.
+- DPC++ compiler is building apps as C++14 apps by default.
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md:518]: @mdtoguchi, is this true for the clang driver? Should we change the implementation to align it with this statement?

Comment (mdtoguchi) [reply sycl/doc/GetStartedGuide.md:518]: Yes, this is true, both Host and Device compilations default to C++14.

Comment (bader) [reply sycl/doc/GetStartedGuide.md:518]: Right... I remember that we removed the custom logic setting C++11 some time ago.
